src/components/RegistrationForm/RegistrationForm.jsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import { Form, Field } from "react-final-form";
+import { useNavigate, Link } from "react-router-dom";
+import "./RegistrationForm.scss";
+import { useDispatch } from "react-redux";
+import { registerUser } from "../../actions/authActions";
+
+const RegistrationForm = () => {
+  const navigate = useNavigate();
+  const dipatch = useDispatch();
+
+  const validate = (values) => {
+    const errors = {};
+    if (!values.userName) {
+      errors.userName = "User Name Required";
+    }
+    if (!values.firstName) {
+      errors.firstName = "First Name Required";
+    }
+    if (!values.lastName) {
+      errors.lastName = "Last Name Required";
+    }
+    if (!values.email) {
+      errors.email = "Email Address Required";
+    } else if (
+      !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)
+    ) {
+      errors.email = "Invalid email address";
+    }
+    if (!values.password) {
+      errors.password = "Password Required";
+    }
+    return errors;
+  };
+
+  const handleRegisterSubmit = (events) => {
+    dipatch(registerUser(events));
+    const existingUsers = JSON.parse(localStorage.getItem("users")) || [];
+    existingUsers.push(events);
+    localStorage.setItem("users", JSON.stringify(existingUsers));
+    navigate("/login");
+  };
```

Comment:
Let's not manage localStorage in component anymore as we currently have store actions for that. Move this to a store action and dispatch the action from here.